string path
uint8 compression_method